gsrs-module-substances-core/src/main/java/gsrs/module/substance/tasks/NameStandardizerTaskInitializer.java

```diff
@@ -0,0 +1,201 @@
+            log.error(e.getMessage());
+            log.error("cannot perform standardization; no standardizer");
+            return;
+        }
+        l.message("Initializing standardization");
+        log.trace("Initializing standardization");
+        //long totalSubstances =substanceRepository.count();
+
+        /*ProcessListener listen = ProcessListener.onCountChange((sofar, total) -> {
+            if (total != null) {
+                l.message("Generated standard names for :" + sofar + " of " + total);
+            } else {
+                l.message("Generated standard names for :" + sofar);
+            }
+        });*/
+
+        l.message("Initializing name standardization: acquiring list");
+
+        /*listen.newProcess();
+        listen.totalRecordsToProcess((int)totalSubstances);
+        log.trace("got list. size: {}", totalSubstances);*/
+
+        ExecutorService executor = BlockingSubmitExecutor.newFixedThreadPool(5, 10);
+        l.message("Initializing name standardization: acquiring user account");
+        Authentication adminAuth = adminService.getAnyAdmin();
+        l.message("Initializing name standardization: starting process");
+        log.trace("starting process");
+
+        try {
+            adminService.runAs(adminAuth, (Runnable) () -> {
+                TransactionTemplate tx = new TransactionTemplate(platformTransactionManager);
+                log.trace("got outer tx " + tx);
+                tx.setPropagationBehavior(TransactionDefinition.PROPAGATION_REQUIRES_NEW);
+                try {
+                    processNames(nameStandardizer, l);
+                } catch (Exception ex) {
+                    l.message("Error standardizing. error: " + ex.getMessage());
+                }
+            });
+        } catch (Exception ee) {
+            log.error("error generating standard names: ", ee);
+            l.message("ERROR:" + ee.getMessage());
+            throw new RuntimeException(ee);
+        }
+
+        l.message("Shutting down executor service");
+        executor.shutdown();
+        try {
+            executor.awaitTermination(1, TimeUnit.DAYS);
+        } catch (InterruptedException e) {
+            //should never happen
+            log.error("Interrupted exception!");
+        }
+        l.message("Task finished");
+        l.complete();
+        //listen.doneProcess();
+    }
+
+    @Override
+    public String getDescription() {
+        return "Regenerate standardized names for all substances in the database";
+    }
+
+    public void setNameStandardizerClassName(String nameStandardizerClassName) {
+        this.nameStandardizerClassName = nameStandardizerClassName;
+    }
+
+    public void setRegenerateNameValue(String regenerateNameValue) {
+        this.regenerateNameValue = regenerateNameValue;
+    }
+
+    private boolean standardizeName(Name name, NameStandardizer nameStandardizer){
```

Comment:
Add a param called "force" which is a boolean. If "force" is true, it should regenerate the stdName even if it's not null/empty (like if the algorithm changes). We sometimes want to fully force regeneration. Will need a config variable for this.

Even when not forced, we want to compare the actual stdName present to the one we _would_ generate. And somehow get a report. Any time that the stdName would be different we should look at them.

Reply:
There's a boolean on the class: forceRecalculationOfAll which does that